R/schema_NamedRange.R

```diff
@@ -0,0 +1,30 @@
+# https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets#NamedRange
+new_NamedRange <- function(namedRangeId,
+                           name,
+                           range) {
+  x <- list(
+    namedRangeId = namedRangeId,
+    name = name,
+    range = range
+  )
+  structure(
+    validate_NamedRange(x),
+    class = "NamedRange"
+  )
+}
+
+validate_NamedRange <- function(x) {
+  # I think read-only vs. required vs. optional status of these elements
+  # depends on what you're trying to do
+  maybe_string(x$namedRangeId, "namedRangeId")
+  maybe_string(x$name, "name")
+
+  # range is an instance of GridRange
+
+  x
+}
+
+NamedRange <- function(...) {
+  x <- new_NamedRange(...)
+  compact(x)
+}
```

Comment:
I used a few variations on the adv-r S3 regime. Where sometimes I call the validator in the low-level constructor and sometimes in the higher-level constructor (all developer facing). The higher-level constructors generally use `compact()` to get rid of what are often a ton of `NULL` elements.